test/powershell/engine/Remoting/InvokeCommandRemoteDebug.Tests.ps1

```diff
@@ -0,0 +1,205 @@
+﻿##
+## PowerShell Invoke-Command -RemoteDebug Tests
+##
+
+if ($IsWindows)
+{
+    if (-not (Get-Module TestRemoting -ErrorAction SilentlyContinue))
+    {
+        $remotingModule = Join-Path $PSScriptRoot "../Common/TestRemoting.psm1"
+        Import-Module $remotingModule
+    }
+
+    $typeDef = @'
+    using System;
+    using System.Globalization;
+    using System.Management.Automation;
+    using System.Management.Automation.Runspaces;
+    using System.Management.Automation.Host;
+
+    namespace TestRunner
+    {
+        public class DummyHost : PSHost, IHostSupportsInteractiveSession
+        {
+            public Runspace _runspace;
+            private Guid _instanceId = Guid.NewGuid();
+
+            public override CultureInfo CurrentCulture
+            {
+                get { return CultureInfo.CurrentCulture; }
+            }
+            public override CultureInfo CurrentUICulture
+            {
+                get { return CultureInfo.CurrentUICulture; }
+            }
+            public override Guid InstanceId
+            {
+                get { return _instanceId; }
+            }
+            public override string Name
+            {
+                get { return "DummyTestHost"; }
+            }
+            public override PSHostUserInterface UI
+            {
+                get { return null; }
+            }
+            public override Version Version
+            {
+                get { return new Version(1, 0); }
+            }
+            public override void EnterNestedPrompt() { }
+            public override void ExitNestedPrompt() { }
+            public override void NotifyBeginApplication() { }
+            public override void NotifyEndApplication() { }
+            public override void SetShouldExit(int exitCode) { }
+            public void PushRunspace(Runspace runspace) { }
+            public void PopRunspace() { }
+            public bool IsRunspacePushed { get { return false; } }
+            public Runspace Runspace { get { return _runspace; } private set { _runspace = value; } }
```

Comment:
We are trying to apply the same standard between test code and product code.

If this code were copied or reused beyond this test (and people definitely do copy our test code - maybe treating it like a sample) - what change would you suggest to avoid problems?

Maybe throw from push/pop - knowing that `Runspace` does not work as intended with push/pop?
At a minimum, I'd say you would add a comment pointing out the flaw *when* the code is used elsewhere.